test/e2e/tools/eks-anywhere-test-tool/pkg/artifacts/artifactFetcher.go

```diff
@@ -71,13 +94,36 @@ func (l *testArtifactFetcher) FetchArtifacts(opts ...FetchArtifactsOpt) error {
 		}
 	}
 
+	var p *cb.Build
+	var err error
+
 	if config.buildId == "" {
-		p, err := l.buildAccountCodebuildClient.FetchLatestBuildForProject(config.project)
+		p, err = l.buildAccountCodebuildClient.FetchLatestBuildForProject(config.project)
 		if err != nil {
 			return fmt.Errorf("failed to get latest build for project: %v", err)
 		}
 		config.buildId = *p.Id
+
+	} else {
+		p, err = l.buildAccountCodebuildClient.FetchBuildForProject(config.buildId)
+		if err != nil {
+			return fmt.Errorf("failed to get build for project: %v", err)
+		}
+	}
+
+	g := p.Logs.GroupName
+	s := p.Logs.StreamName
+
+	logs, err := l.buildAccountCwClient.GetLogs(*g, *s)
+	if err != nil {
+		return fmt.Errorf("fetching cloudwatch logs: %v", err)
+	}
+
+	failedTests, err := filterFailedTests(logs)
+	if err != nil {
+		return err
 	}
+	failedTestsMap := testResultMap(failedTests)
```

Comment:
I think it would be useful to have a single public method, such as `GetFailedTests`, which we could use across the tool to do this exact routinue